File: March 2026 GFG SOLUTION/March-04.py
class Solution:
    def maxSubarrayXOR(self, arr, k):
        xr = res = 0
        for i in range(len(arr)):
            xr ^= arr[i]
            if i >= k:
                xr ^= arr[i - k]
            if i >= k - 1:
                res = max(res, xr)
        return res

File: March 2026 GFG SOLUTION/March-04.java
class Solution {
    public int maxSubarrayXOR(int[] arr, int k) {
        int xr = 0, res = 0;
        for (int i = 0; i < arr.length; i++) {
            xr ^= arr[i];
            if (i >= k) xr ^= arr[i - k];
            if (i >= k - 1) res = Math.max(res, xr);
        }
        return res;
    }
}
